daliuge-translator/dlg/dropmake/cwl.py

```diff
@@ -28,7 +28,10 @@
 import cwlgen
 
 from dlg import common
+from dlg.common import Categories
 
+# the following node categories are not supported by the CWL translator
+UNSUPPORTED_CATEGORIES = [Categories.COMPONENT, Categories.MPI, Categories.DYNLIB_APP, Categories.DYNLIB_PROC_APP, Categories.DOCKER]
```

Comment:
The list of unsupported categories is way longer (and might grow) compared to the list of *supported* categories (only one). Maybe the logic could be inverted to list the supported categories, then do `dataType not in SUPPORTED_CATEGORIES` below.

Reply:
Agreed, I'll make this change. The list of supported categories is just BASH_SHELL_APP and FILE.